### PR TITLE
Implement json un-/marshalling for gcs.Path.

### PR DESCRIPTION
**Pull request by fejta**
This allows logrus' json formatter to provide a useful string.
Other cosmetic fixes

### CODE REVIEW

File: util/gcs/gcs.go [hunk 1]
```diff
@@ -107,6 +108,22 @@ func (g *Path) SetURL(u *url.URL) error {
 	return nil
 }
 
+func (g Path) MarshalJSON() ([]byte, error) {
+	return json.Marshal(g.String())
+}
+
+func (g *Path) UnmarshalJSON(buf []byte) error {
```
Comment (chases2) [inline util/gcs/gcs.go:115]: A function comment here that indicates that this is for logrus logging of the path itself (and not, say, reading the file at `Path` and unmarshaling that) would be nice

/lint

**Review by chases2 — APPROVED** [review]
Documentation nits from the linter only. Thanks for improving our logging!

/hold for author review

File: util/gcs/gcs.go [hunk 2]
```diff
@@ -107,6 +108,22 @@ func (g *Path) SetURL(u *url.URL) error {
 	return nil
 }
 
+func (g Path) MarshalJSON() ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (fejta) [reply util/gcs/gcs.go:112]: Done

File: util/gcs/gcs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fejta) [reply util/gcs/gcs.go:117]: Done

Comment (fejta) [reply util/gcs/gcs.go:117]: Done

**Review by michelle192837 — APPROVED** [review]

**Comment by fejta** [comment]
oh LOL I held it
/hold cancel

**Pull request merged**